code/mapping/ext_modules/mapping_common/transform.py

```diff
@@ -0,0 +1,305 @@
+from dataclasses import dataclass
+
+import numpy as np
+import numpy.typing as npt
+import math
+
+from geometry_msgs import msg as geometry_msgs
+from mapping import msg
+
+
+@dataclass(init=False, eq=False)
+class _Coord2:
+    """Homogenous 2 dimensional coordinate
+
+    This base class should be abstract,
+    but cython does not support the ABC superclass and decorators
+    """
+
+    # Matrix with shape (3)
+    _matrix: npt.NDArray[np.float64]
+
+    def __init__(self, matrix: npt.NDArray[np.float64]) -> None:
+        assert matrix.shape == (
+            3,
+        ), f"{type(self).__name__} matrix must have shape (3,)"
+        self._matrix = matrix
+
+    def x(self) -> float:
```

Comment:
mark this with @ property. then we write coord.x afterwards instead of coord.x() which is much more readable

same for y as well

Reply:
Would be nice, but cython currently does not properly support the property decorator :-(